shared/src/commonMain/kotlin/io/customer/shared/database/QueryHelper.kt

```diff
@@ -0,0 +1,264 @@
+package io.customer.shared.database
+
+import com.squareup.sqldelight.TransactionWithReturn
+import io.customer.shared.common.QueueTaskResult
+import io.customer.shared.common.Success
+import io.customer.shared.runSuspended
+import io.customer.shared.sdk.config.BackgroundQueueConfig
+import io.customer.shared.sdk.meta.Workspace
+import io.customer.shared.tracking.constant.Priority
+import io.customer.shared.tracking.constant.QueueTaskStatus
+import io.customer.shared.tracking.model.*
+import io.customer.shared.tracking.queue.TaskResultListener
+import io.customer.shared.tracking.queue.failure
+import io.customer.shared.tracking.queue.success
+import io.customer.shared.util.*
+import io.customer.shared.work.BackgroundDispatcher
+import io.customer.shared.work.MainDispatcher
+import kotlinx.datetime.Instant
+import local.TrackingTask
+import local.TrackingTaskQueries
+
+/**
+ * The class works as a bridge for SQL queries. All queries to database should be made using this
+ * class to provide abstraction and keep communication with database easier.
+ */
+internal interface QueryHelper {
+    @MainDispatcher
+    fun insertTask(
```

Comment:
I think we can instead rename the class to `TrackingTaskQueryHelper` where `TrackingTask` is the name of our DB table. This can give more information on what the task will be. Also, by following this pattern, we can create a `QueryHelper` for every primary table we have in our SDK.

Reply:
Since I don't see any of the db code being used in the background queue implementation, I am not sure what the best naming is or the DB schema to use. I want to ask why we would need more then 1 DB table since I think the only thing we are saving into the DB right now is background queue tasks. I would think that 1 table could keep the DB layer simple to just save background queue tasks. 

I think you're suggesting having `TrackingTask` just for tracking tasks? Then make more tables in the future for other types of BQ tasks?

Reply:
No, we don't plan to have any other kind of task right now. But `Tracking` was appended to database model as we already have `Task` class and wanted to use a different name. Let's just keep it this way then, I'll add this to list of unresolved conversations and we can then rename it if needed after merging BQ changes so you have clear picture of classes and their usage when suggesting.